atecc608a/atecc608a_utils.c

```diff
@@ -43,6 +43,67 @@ psa_status_t atecc608a_get_serial_number(uint8_t *buffer,
     return status;
 }
 
+static uint16_t getCrc16(const uint8_t *data, uint8_t length)
+{
+    /* The code below is from
+     * Atmel-8936A-CryptoAuth-Data-Zone-CRC-Calculation-ApplicationNote_082015
```

Comment:
There is no explicit license listed. The only information that is present in the application note is as follows:

> DISCLAIMER: The information in this document is provided in connection with Atmel products. No license, express or implied, by estoppel or otherwise, to any intellectual property right
is granted by this document or in connection with the sale of Atmel products. EXCEPT AS SET FORTH IN THE ATMEL TERMS AND CONDITIONS OF SALES LOCATED ON THE ATMEL
WEBSITE, ATMEL ASSUMES NO LIABILITY WHATSOEVER AND DISCLAIMS ANY EXPRESS, IMPLIED OR STATUTORY WARRANTY RELATING TO ITS PRODUCTS INCLUDING,
BUT NOT LIMITED TO, THE IMPLIED WARRANTY OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, OR NON-INFRINGEMENT. IN NO EVENT SHALL ATMEL BE
LIABLE FOR ANY DIRECT, INDIRECT, CONSEQUENTIAL, PUNITIVE, SPECIAL OR INCIDENTAL DAMAGES (INCLUDING, WITHOUT LIMITATION, DAMAG ES FOR LOSS AND
PROFITS, BUSINESS INTERRUPTION, OR LOSS OF INFORMATION) ARISING OUT OF THE USE OR INABILITY TO USE THIS DOCUMENT, EVEN IF ATMEL HAS BEEN ADVISED
OF THE POSSIBILITY OF SUCH DAMAGES. Atmel makes no representations or warranties with respect to the accuracy or completeness of the contents of this document and reserves
the right to make changes to specifications and products descriptions at any time without notice. Atmel does not make any commitment to update the information contained herein.

Reply:
I am not a lawyer, but I find this part a little bit ominous:

> No license, express or implied, by estoppel or otherwise, to any intellectual property right
is granted by this document

Reply:
As discussed in private with Jaeden, I'll use a builtin crc function from the cryptoauthlib.